pkg/controller/volume/pvprotection/pv_protection_controller.go

```diff
@@ -69,11 +69,37 @@ func NewPVProtectionController(pvInformer coreinformers.PersistentVolumeInformer
 		UpdateFunc: func(old, new interface{}) {
 			e.pvAddedUpdated(new)
 		},
+		DeleteFunc: func(obj interface{}) {
```

Comment:
I'm a little confused.  If PV object is deleted, then there is no more PV object to remove the finalizer on?

Reply:
Please look at line 83 below - we obtain the PV from tombstone.

Reply:
But I'm not sure if it's actually possible to update an object (that removes the finalizer) that has been deleted already.

Reply:
@msau42 
Looking at Controller#removeFinalizer, removing finalizer from pvClone.ObjectMeta.Finalizers is feasible in this case.

Reply:
@msau42 
Please take a look at pkg/controller/service/service_controller.go , around line 310.